go/mysql/server.go

```diff
@@ -312,32 +323,86 @@ func (l *Listener) Addr() net.Addr {
 
 // Accept runs an accept loop until the listener is closed.
 func (l *Listener) Accept() {
-	for {
+	var sem chan struct{}
+	if l.maxConns > 0 {
+		sem = make(chan struct{}, l.maxConns)
+	}
+
+	waitingConnections := sync2.NewAtomicInt64(0)
```

Comment:
Maybe prefer just `var waitingConnections atomic.Int64` from `sync/atomic`. `sync2` predates `sync/atomic.{Ui,I}nt{32,64}` and friends...